### PR TITLE
Portfolio tests

### PR DESCRIPTION
**Pull request by peterpie123**
Create unit tests for Java backend. JavaScript testing doesn't play well with DOM interaction, so it has been omitted.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/CommentAnalysis.java [hunk 1]
```diff
@@ -74,26 +74,36 @@ public CommentAnalysis() {
...
       List<EntityAnnotation> labels = getImageLabels(imageBytes);
       labels.stream().forEach(entity -> imageLabels.add(new ImageLabel(entity)));
     }
-    comment.getBlobKey().ifPresent(blobKey -> {
-      
-    });
+  }
+
+  /** Same as analyzeText, just with a configurable GCloud api */
+  void analyzeText(Comment comment, LanguageServiceClient client) {
```
[the remark this answers is not in the document]
Comment (peterpie123) [reply portfolio/src/main/java/com/google/sps/data/CommentAnalysis.java:85]: I use it to facilitate dependency injection for testing - that's why it's package private.

File: portfolio/package.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "scripts": {
+    "test": "mocha --recursive ./src"
+  }
+}
```
Comment (davidgaskins) [inline portfolio/package.json:5]: new line please :)

File: portfolio/package.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "scripts": {
+    "test": "mocha --recursive ./src"
```
Comment (davidgaskins) [inline portfolio/package.json:3]: per PR description, js tests are omitted. Should this file then also be omitted from this PR?

File: portfolio/src/main/java/com/google/sps/data/CommentAnalysis.java [hunk 4]
```diff
@@ -45,9 +45,9 @@
   private float sentimentScore;
 
   /** Represents a single label for an image. Serves as a convenient wrapper for serialization. */
-  private static class ImageLabel {
+  public static class ImageLabel {
```
Comment (davidgaskins) [inline portfolio/src/main/java/com/google/sps/data/CommentAnalysis.java:48]: should this class go into its own file now?

File: portfolio/src/test/java/com/google/sps/servlets/AnalyzeServletTest.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    when(request.getParameter("id")).thenReturn("" + id);
+    when(helper.getCommentById(id)).thenReturn(Optional.empty());
+
+    servlet.doPost(request, response, analysis);
+
+    verify(analysis, never()).analyzeText(any(Comment.class));
+    verify(analysis, never()).analyzeImage(any(Comment.class));
+    verify(response, never()).getWriter();
+  }
+}
```
Comment (davidgaskins) [inline portfolio/src/test/java/com/google/sps/servlets/AnalyzeServletTest.java:84]: new line please :)

File: portfolio/src/test/java/com/google/sps/servlets/ImageServletTest.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** */
```
Comment (davidgaskins) [inline portfolio/src/test/java/com/google/sps/servlets/ImageServletTest.java:30]: this Javadoc can be removed

File: portfolio/src/test/javascript/test-comments.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+import { equal } from 'assert';
+
+describe('Utilities', () => {
+  describe('animateElement', () => {
+    equal(true, true);
+  })
+});
```
Comment (davidgaskins) [inline portfolio/src/test/javascript/test-comments.js:7]: newline please :)

**Pull request merged**